course-design.md

```diff
@@ -0,0 +1,214 @@
+            - Maybe provide several potential problem sets as starting points? Or 
+            get the learners to determine their own? (this might be difficult) 
+            Optional is to choose a few simple existing packages (that don't
+            follow these guidelines from the book) and get them to re-make it
+            using the principles (and maybe later ask package maintainer if they
+            would like the features as a PR?)
+        - Formative assessments like coding with immediate feedback on correct/incorrect
+        (via learnr package? Esp with new update to RStudio in version 1.3)
+          - Need to consider that writing correctness tests is time consuming
+        - Reading assignments to motivate and inform on content given in-class
+
+- **Assessment and feedback activities**:
+    - Deadlines for milestones to achieve for projects, with instructor feedback on progress
+    - Out-of-class formative assessments based on maybe learnr tutorials?
+    - After completing in-class exercises, go over them as class?
+    - Peer review activities? E.g. exercises that have a rubric that go with them that can be used to guide peer review.
+    - Assigned TA for group to get feedback on group work project?
+    - Contribute to existing packages on GitHub by submitting PR
+
+- **Content for the instructor and teaching assistant**
+    - Such as guidelines and suggestions on what do to at each portion of the course
+    for both instructor and teaching assistant
+    - A TA could be assigned to a group and be the learners direct link to getting feedback
+    and help
+    - TA could act as Carpentry-style helpers during the live coding sessions
+    - Guidelines on how to make this course either online or in person
+
+- Key resources to use for this:
+  - R packages: usethis, devtools, drake?, pkgdown, testthat, styler, rmarkdown
+  - Online books: [R Packages](https://r-pkgs.org/) 
+  (we differentiate from this amazing book by structuring everything around 
+  this being a course, with exercises, assessments, (group?) projects, 
+  instructions/comments/notes for instructors, etc)
+    - We use the R Packages book as a reference
+    
+## Chapter outline
+
+Course chapters and content need to be strongly aligned with learning outcome/goal.
+Here is a draft outline to guide development.
+
+- Overview of material
+    - How to use, why to use, general details and overall description
+        - Reference "For Instructors" section for instructors to check out
+
+- Syllabus
+    - Overview of course
+    - Why to learn it
+    - Intended learner (personas, assumptions and expectations etc)
+    - What makes this course different
+        - Go over the "Whole Game" chapter of the R Packages book (https://r-pkgs.org/whole-game.html)
+        to give an idea of what it all looks like in the end of this course
+
+- Folder and file structure (RStudio R Projects)
+    - File paths, very basics of the shell (in RStudio)
```

Comment:
`here` and setting the working directory

Reply:
That might be ok to talk about but not necessarily in the package context. Package set up and building already implicitly use a standard way of interacting with working directory etc, so that might not make sense for this book.